### PR TITLE
feat(dealing with multiple graphs / making compatible with current text2graph)

### PR DESCRIPTION
**Pull request by zachares**

### CODE REVIEW

File: src/transformers/models/processing_graphs_within_model/causal_message_passing.py [hunk 1]
```diff
@@ -189,17 +189,19 @@ def add_edge(
...
             end_idx=end_idx,
             message_passing_dict=message_passing_dict
         )
-        curr_length = len(message_passing_dict[f"tokens2elements"])
+        curr_length = len(message_passing_dict["tokens2elements"])
         if curr_length > prev_length:
             current_idx = len(message_passing_dict["tokens2elements"]) - 1
-            node2edge_idxs[pred_node.ids].append(current_idx)
-            node2edge_idxs[succ_node.ids].append(current_idx)
+            pred_ids = pred_node.graph_id + pred_node.ids
```
Comment (vahanhov) [inline src/transformers/models/processing_graphs_within_model/causal_message_passing.py]: do we need a special token between graph_id and predecessor ids?

**Review by vahanhov — APPROVED** [review]
just one little comment

File: src/transformers/models/processing_graphs_within_model/causal_message_passing.py (hunk 1 above)
Comment (zachares) [reply src/transformers/models/processing_graphs_within_model/causal_message_passing.py:201]: no

Comment (zachares) [reply src/transformers/models/processing_graphs_within_model/causal_message_passing.py]: that is a good idea, just to deal with a case which would be very unlikely but possible!

File: src/transformers/models/processing_graphs_within_model/causal_message_passing.py [hunk 2]
```diff
@@ -172,12 +171,13 @@ def add_node(
             message_passing_dict=message_passing_dict
         )
         curr_length = len(message_passing_dict[f"tokens2elements"])
-        if last_occurence_idx[current_occurence.ids] != -1 and curr_length > prev_length:
+        full_ids = current_occurence.graph_id + current_occurence.ids
```
Comment (zachares) [inline src/transformers/models/processing_graphs_within_model/causal_message_passing.py]:
```suggestion
        full_ids = current_occurence.graph_id + ("--node_ids--",) + current_occurence.ids
```

File: src/transformers/models/processing_graphs_within_model/causal_message_passing.py (hunk 1 above)
Comment (zachares) [inline src/transformers/models/processing_graphs_within_model/causal_message_passing.py]:
```suggestion
            pred_ids = pred_node.graph_id + ("--node_ids--",) + pred_node.ids
```

File: src/transformers/models/processing_graphs_within_model/causal_message_passing.py [hunk 3]
```diff
@@ -189,17 +189,19 @@ def add_edge(
...
         )
-        curr_length = len(message_passing_dict[f"tokens2elements"])
+        curr_length = len(message_passing_dict["tokens2elements"])
         if curr_length > prev_length:
             current_idx = len(message_passing_dict["tokens2elements"]) - 1
-            node2edge_idxs[pred_node.ids].append(current_idx)
-            node2edge_idxs[succ_node.ids].append(current_idx)
+            pred_ids = pred_node.graph_id + ("--node_ids--",) + pred_node.ids
+            node2edge_idxs[pred_ids].append(current_idx)
+            succ_ids = succ_node.graph_id + succ_node.ids
```
Comment (zachares) [inline src/transformers/models/processing_graphs_within_model/causal_message_passing.py]:
```suggestion
            succ_ids = succ_node.graph_id + ("--node_ids--",) + succ_node.ids
```

**Pull request merged**